brion/plugin/morphologySWC.cpp

```diff
@@ -80,15 +76,24 @@ struct Sample
     bool valid;
     Vector4f point; // x, y, z and diameter
     SWCSectionType type;
-    int parent;
-    int nextID;
-    int siblingID;
-    int parentSection; // Only meaningful for the first sample of each
-                       // section
+    int parent = -1;        // sample index
+    int nextID = -1;        // sample index
+    int siblingID = -1;     // sample index
+    int sectionID = -1;     // Only necessary for the last sample of a section
+    int parentSection = -1; // Only meaningful for the first sample of a section
+};
+using Samples = std::vector<Sample>;
+auto _comparator = [](const Sample* a, const Sample* b) {
+    // We want undefined sections to go last but we need to respect the
+    // numerical values from the swc "specification".
+    // The conversion of -1 to size_t takes care of this
```

Comment:
uint32_t?

Neat trick, but I wonder if would be better to be explicit about handling SWCSectionType::SWC_SECTION_UNDEFINED, that way it's safe against refactors and such.

Reply:
The alternative would be to add if's to the code and they make me sad.
I think the current code is safe against refactorings because, as the comment states, the numeric values of the enum values come from the SWC spec and we can't change that. Otherwise I would have moved UNDEFINED to another position in the enum declaration and done.